traffic_prophet/countmatch/permcount.py

```diff
@@ -0,0 +1,147 @@
+"""Sift and process permanent counts."""
+
+import warnings
+from tqdm.auto import tqdm
+
+from . import base
+from . import derivedvals as dv
+from . import growthfactor as gf
+from .. import cfg
+
+
+class PermCount(base.Count):
+    """Class to hold permanent count data and calculate growth factors.
+
+    Parameters
+    ----------
+    count_id : int
+        ID of count.
+    centreline_id : int
+        Centreline ID associated with count.
+    direction : +1 or -1
+        Direction of traffic travel.
+    data : pandas.DataFrame
+        Raw daily traffic counts.
+    perm_years : list
+        List of years to use as permanent count.
+    """
+
+    def __init__(self, count_id, centreline_id, direction, data,
+                 perm_years):
+        super().__init__(count_id, centreline_id, direction, data,
+                         is_permanent=True)
+        if len(perm_years) == 0:
+            raise ValueError("cannot have empty perm_years!")
+        self._perm_years = perm_years
+        self._growth_fit = None
+
+    @property
+    def perm_years(self):
+        return self._perm_years
+
+    @property
+    def growth_factor(self):
+        if self._growth_fit is None:
+            raise AttributeError("PTC has not had its growth factor fit!")
+        return self._growth_fit['growth_factor']
+
+    @classmethod
+    def from_count_object(cls, tc, perm_years):
+        return cls(tc.count_id, tc.centreline_id, tc.direction, tc.data,
+                   perm_years)
+
+
+class PermCountProcessor:
+    """Class for processing a list of counts into PTCs and STTCs.
+
+    Currently a permanent count needs to have at least one year with all 12
+    months and a sufficient number of total days represented, not be from HW401
+    and not be excluded by the user in the config file.
+
+    Parameters
+    ----------
+    dv_calc : DerivedVal instance
+        For imputation and derived properties.
+    gf_calc : GrowthFactor instance
+        For estimating growth factor.
+    cfg : dict, optional
+        Configuration settings.  Default: `config.cm`.
+    """
+
+    def __init__(self, dv_calc, gf_calc, cfg=cfg.cm):
+        self.dvc = dv_calc
+        self.gfc = gf_calc
+        self.cfg = cfg
+        # Obtain a list of count_ids that (according to TEPs-I) should not be
```

Comment:
`Obtain a list of `count_id`s that should be excluded from being PTCs.`  No need to discuss TEPs if we eventually customize this list.